BEPs/BEP-319.md

```diff
@@ -0,0 +1,97 @@
+</div>
+The existing allocation steps are as follows:
+
+1. Validators collect transaction fees from within the block.
+2. In the client, if the balance of the SystemRewardContract is less than 100 BNB, 1/16 of the collected transaction fees is allocated as rewards for relayers and Fast Finality voting. The remainder is deposited into the ValidatorContract.
+3. Within the ValidatorContract, 10% of the deposited amount is burned directly, and the remainder serves as potential earnings for validators and stakers.
+
+#### 4.1.2 Allocation Logic Moved to the Contract 
+<div align="center">
+<img src=./assets/BEP-319/4.2-new-fee-distribution.png width=90% />
+</div>
+Due to the deposition of the SystemRewardContract being in the client, requiring hard forks for each change, and for ease of governance, it is moved to the smart contract. The adjusted allocation logic is as follows:
+
+1. Validators collect transaction fees from within the block and deposit all into the ValidatorContract.
+2. Deposit systemRewardRatio of transaction fees into the SystemRewardContract, where systemRewardRatio is governable, with an initial value of 1/16.
+3. Within the ValidatorContract, 9.375% of the deposited amount is burned directly, and the remainder serves as potential earnings for validators and stakers.
+#### 4.1.3 Maintain Burn Ratio 
+The real-time burning mechanism is an essential part of the BNB deflation plan. In this BEP, the goal is to maintain it essentially unchanged.
+Before implementing this BEP, a portion of the balance of the SystemRewardContract will be used as a reward for Fast Finality voting at the end of each epoch, resulting in a balance less than 100 BNB. Therefore, in the majority of blocks, validators will deposit 1/16 of the block transaction fees to the SystemRewardContract.
+The oldBurnRatio is 10%. The actual proportion of burned transaction fees to block transaction fees can be calculated approximately as follows:
+
+```
+newBurnRatio = (1 - 1/16) * oldBurnRatio = (1 - 1/16) * 10% = 9.375%
+```
+As a result, following the implementation of this BEP, the burn ratio is set to 9.375% directly.
+## 4.2 Balancing Rewards
+Fast Finality rewards are unevenly distributed between epochs. Assuming the current balance of the systemReward contract is currentSystemRewardBalance, and the balance after award distribution in the previous epoch is previousSystemRewardBalance, the totalReward calculation formula is as follows:
+```
+if currentSystemRewardBalance > 100BNB
+  totalReward = currentSystemRewardBalance / 100
+else
+  totalReward = (currentSystemRewardBalance - previousSystemRewardBalance) * 0.5
+```
+This algorithm allocates a portion of the systemReward contract balance increment for each epoch, causing the systemReward contract balance to continuously grow. When it exceeds 100BNB, a reward exceeding 1 BNB is distributed at once, far more than other epochs. 
+The new formula is as follows:
+```
+if currentSystemRewardBalance > 100BNB
```

Comment:
What about the case when `currentSystemRewardBalance < 100 BNB` ?

Reply:
![image](https://github.com/bnb-chain/BEPs/assets/7995985/22553a1e-0c36-44a9-8a82-8d5b51c952b7)

Reply:
It rarely happen

Reply:
updated

Reply:
Under what conditions will the balance drop below 100BNB and what is estimated likelihood of this happening? I think it would be good to have an analysis of that before going ahead with this BEP, because if the balance drops below 100BNB there will be no finality rewards, and thus no incentive for validators to send their fast finality votes.

Reply:
nearly 100%, 
only happens when bsc-relayers got some errors sending crosschain messages to bsc 
and retry again and again, then a lot of balance of systemReward will need to pay to the relayers.

this rarely happend, in the past year only once as I know

Reply:
Is there a reason why the balance is always 100BNB and above?  I don't know if it's a good idea to base the logic to fit the historical state of the chain. This makes it hard to test this BEP on other networks like a local devnet or even the testnet.